unlock-app/src/unlockTypes.ts

```diff
@@ -4,54 +4,197 @@
+    }),
+    defaultValue: z
+      .string({
+        description: 'Field default value',
+      })
+      .optional(),
+    type: z.enum(['text', 'date', 'color', 'email', 'url']),
+    required: z.boolean({
+      description: 'When true the field will be required',
+    }),
+    placeholder: z
+      .string({
+        description: 'Field placeholder text',
+      })
+      .optional(),
+    public: z
+      .boolean({
+        description:
+          'If any metadata should be visible to everyone, mark the public field as true.',
+      })
+      .optional(), // optional, all non-public fields are treated as protected
+  })
+  .array()
+  .optional()
 
 export const PaywallConfigLockSchema = z.object({
-  name: z.string().optional(),
+  name: z
+    .string({
+      description: 'Name of the lock to display.',
+    })
+    .optional(),
   network: z.number().int().positive().optional(),
-  metadataInputs: z.array(MetadataInputSchema).optional(),
-  recurringPayments: z.union([z.string(), z.number()]).optional(),
-  captcha: z.boolean().optional(),
-  password: z.boolean().optional(),
-  emailRequired: z.boolean().optional(),
-  maxRecipients: z.number().int().positive().optional(),
-  minRecipients: z.number().int().positive().optional(),
-  superfluid: z.boolean().optional(),
+  metadataInputs: MetadataInputSchema,
+  recurringPayments: z
+    .number({
+      description:
+        'The number of time a membership should be renewed automatically. This only applies to ERC20 locks.',
+    })
+    .int()
+    .optional(),
+  captcha: z
+    .boolean({
+      description:
+        'If set true, the users will be prompted to go through a captcha during the checkout process. This is better used in conjunction with a purchase hook that verifies that captcha is valid.',
+    })
+    .optional(),
+  password: z
+    .boolean({
+      description:
+        'Defaults to false. If set to true, the user will be prompted to enter a password in order to complete their purchases. This will only be useful if the lock is connected to a hook that will handle the password verification.',
+    })
+    .optional(),
+  emailRequired: z
+    .boolean({
+      description:
+        'If set to true, the user will be prompted to enter an email which will be stored as metadata and be visible to any lock manager.',
+    })
+    .optional(),
+  minRecipients: z
+    .number({
+      description:
+        'Set the minimum number of memberships a user needs to purchase.',
+    })
+    .int()
+    .positive()
+    .optional(),
+  maxRecipients: z
+    .number({
+      description: `Set the max number of memberships a user can purchase. Note: By default, checkout doesn't allow fiddling with quantity. You have to set maxRecipients to allow for changing to quantity.`,
+    })
+    .int()
+    .positive()
+    .optional(),
+  superfluid: z
+    .boolean({
+      description:
+        'When set to true, superfluid will be enabled as payment method for the lock.',
```

Comment:
@julien51 have any better description here? when don't have description in docs. also we need to add it there?

Reply:
HA! You merged already so I can't update :/